cg/apps/lims/api.py

```diff
@@ -478,3 +484,47 @@ def get_latest_rna_input_amount(self, sample_id: str) -> float | None:
         )
         input_amount: float | None = self._get_last_used_input_amount(input_amounts=input_amounts)
         return input_amount
+
+    def get_latest_artifact_for_sample(
+        self, process_type: LimsProcess, artifact_type: LimsArtifactTypes, sample_internal_id: str
+    ) -> Artifact:
+        """Return latest artifact for a given sample, process and artifact type."""
+
+        artifacts: list[Artifact] = self.get_artifacts(
+            process_type=process_type,
+            type=artifact_type,
+            samplelimsid=sample_internal_id,
+        )
+        try:
+            return artifacts[0]
+        except IndexError as error:
+            raise LimsDataError() from error
+
+    def get_internal_negative_control_id_from_sample_in_pool(
+        self, sample_internal_id: str, pooling_step: LimsProcess
+    ) -> str:
+        """Retrieve from lims the sample_id for the internal negative control sample present in the same pool as the given sample."""
+        artifact: Artifact = self.get_latest_artifact_for_sample(
+            process_type=pooling_step,
+            artifact_type=LimsArtifactTypes.ANALYTE,
```

Comment:
`LimsArtifactTypes.ANALYTE` seems like the default value. If that's the case, consider defining `artifact_type=LimsArtifactTypes.ANALYTE` directly in the arguments of `get_latest_artifact_for_sample`, and remove it from here.

Reply:
Changed this, but I'm actually a bit unsure if the benefit is worth it if the function is to be as general as possible.